apps/frontend/src/lib/message/tapBack.ts

```diff
@@ -0,0 +1,70 @@
+import { z } from "zod";
+import { randomBytes } from "crypto";
+import { getCounterMessage, sign } from "../crypto/babyJubJub";
+import { ChipIssuer } from "@types";
+import {
+  TapDataSchema,
+  MessageType,
+  MessageTypeSchema,
+  UserDataSchema,
+  UserData,
+} from "../storage/types";
+
+export const TapBackMessageSchema = z.object({
+  type: MessageTypeSchema,
+  user: UserDataSchema,
+  tap: TapDataSchema,
+});
+
+export type TapBackMessage = z.infer<typeof TapBackMessageSchema>;
+
+export interface GenerateTapBackMessageArgs {
+  user: UserData;
+  chipPrivateKey: string;
+  chipPublicKey: string;
+  chipIssuer: ChipIssuer;
+}
+
+export const generateSerializedTapBackMessage = async ({
+  user,
+  chipPrivateKey,
+  chipPublicKey,
+  chipIssuer,
+}: GenerateTapBackMessageArgs): Promise<{
+  serializedMessage: string;
+  messageTimestamp: Date;
+}> => {
+  const msgNonce = 0; // No counter for tap backs (4 bytes)
+  const generatedRandomBytes = randomBytes(28); // 28 random bytes
+  const message = getCounterMessage(
+    msgNonce,
+    generatedRandomBytes.toString("hex")
+  );
```

Comment:
Comprehension question -- will the message be the same for all tap backs? And if so, what is its role?